app/assets/javascripts/general.js

```diff
@@ -65,4 +65,54 @@ $(document).ready(function() {
     $this.unbind("mouseup");
     return false;
   });
+
+  $('.js-toggle-delivery-log').on('click', function(e){
```

Comment:
I resisted adding logic for this, partly because it's get another string that needs to be translated, but mostly because this is just placeholder wording, and the end goal is that the link text should be the message status like "delivered", rather than "show/hide".